gnosis-imap/gnosis-imap-spec.ini

```diff
@@ -148,6 +148,158 @@ proxy_storage:
 +requires:
     andBool 0 <Int DATA_LEN
 
+[removeOwner-success]
+k: (#execute => #halt) ~> _
+output: _ => _
+statusCode: _ => EVMC_SUCCESS
+callStack: _
+this: #PROXY_ID
+msg_sender: MSG_SENDER
+callData: #abiCallData("removeOwner", (#address(PREV_OWNER), 
+                                       #address(OWNER), 
+                                       #uint256(NEW_THRESHOLD)))
+callValue: 0
+wordStack: .WordStack => _
+localMem: .IMap => _
+pc: 0 => _
+gas: #gas(STARTGAS, 0, 0) => _
+memoryUsed: 0 => _
+refund: _ => _
+coinbase: _ => _
+proxy_storage:
+    store(store(store(store(M1:IMap, #hashedLocation({COMPILER}, {OWNERS}, OWNER), INIT_OWNER),
+                                     #hashedLocation({COMPILER}, {OWNERS}, PREV_OWNER), INIT_PREV_OWNER),
+                                     #hashedLocation({COMPILER}, {THRESHOLD}, .IntList), THRESHOLD),
+                                     #hashedLocation({COMPILER}, {OWNERCOUNT}, .IntList), OWNERCOUNT)
+    => M2:IMap
++requires:
+    ; Address public constant SENTINEL_OWNERS = address(0x1);
+    andBool SENTINEL ==Int 1
+    ; Range
+    andBool #rangeAddress(  PREV_OWNER)
+    andBool #rangeAddress(  OWNER)
+    andBool #rangeAddress(  INIT_OWNER)
+    andBool #rangeAddress(  INIT_PREV_OWNER)
+    andBool #rangeUInt(256, NEW_THRESHOLD)
+    andBool #rangeUInt(256, THRESHOLD)  
+    andBool #rangeUInt(256, OWNERCOUNT)
+    andBool #rangeUInt(256, OWNERCOUNT -Int 1)
```

Comment:
```suggestion
    ; Contract invariant
    andBool OWNERCOUNT >=Int 1
```

Reply:
The range pre-condition should be limited to the one derived by its type.

This condition is different. It is rather an invariant maintained by the entire contract. Reasoning about the invariant can be done separately. Let's mark this invariant condition clearly, and come back to this later once we prove all others.